### DISCUSSION TITLE
why would you

### DISCUSSION THREAD

**Discussion by EnetoJara** [lmaccherone/node-localstorage#78] [Q&A]
I do not want to disrespect, but honestly. maybe I am to dumb to see the magnificent idea

**Comment by trasherdk** [comment]
Convenience.

Don't want to run a Redis, MySQL or other DB server, for a key/value storage.

Testing browser javascript in Node (Mocha/Jest/Whatever).

Probably other use cases too :smiley:

**Comment by lmaccherone** [comment]
Those are the two primary use cases but the first one has two sub-motivations: 1) lightweight, and 2) write code that will run on both the browser and the server.

That said, I think the majority of uses are for testing

**Comment by lmaccherone** [comment]
Note, it's also synchronous which simplifies coding
